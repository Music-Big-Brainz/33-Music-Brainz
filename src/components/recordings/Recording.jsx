import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const Recording = ({ recordingId, songTitle, name, albumId }) => {
  return (
    <Link to={`/releases/${name}/title/${albumId}/${songTitle}/${songTitle}`}>
      <figure key={recordingId}>
        <p>{songTitle}</p>
      </figure>
    </Link>
  );
};

Recording.propTypes = {
  name: PropTypes.string.isRequired,
  albumId: PropTypes.string.isRequired,
  recordingId: PropTypes.string.isRequired,
  songTitle: PropTypes.string.isRequired,
};

export default Recording;
