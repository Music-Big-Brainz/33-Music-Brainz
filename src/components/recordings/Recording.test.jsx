import React from 'react';
import { render } from '@testing-library/react';
import Recording from './Recording';
import { MemoryRouter } from 'react-router-dom';

describe('displays an album', () => {
  it('renders one single album', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Recording
          recordingId="recordingId"
          songTitle="songTitle"
          name="name"
          albumId="albumbId"
        />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
