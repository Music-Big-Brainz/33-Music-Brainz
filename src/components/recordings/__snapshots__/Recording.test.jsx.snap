// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays an album renders one single album 1`] = `
<DocumentFragment>
  <a
    href="/releases/name/title/albumbId/songTitle/songTitle"
  >
    <figure>
      <p>
        songTitle
      </p>
    </figure>
  </a>
</DocumentFragment>
`;
